lib/puppetserver/ca/utils/config.rb

```diff
@@ -22,39 +22,40 @@ def self.munge_alt_names(names)
         end
 
         def self.puppet_confdir
-          if self.running_as_root?
+          if running_as_root?
             '/etc/puppetlabs/puppet'
           else
             "#{ENV['HOME']}/.puppetlabs/etc/puppet"
           end
         end
 
-        def self.puppetserver_confdir
-          if self.running_as_root?
-            '/etc/puppetlabs/puppetserver'
-          else
-            "#{ENV['HOME']}/.puppetlabs/etc/puppetserver"
-          end
+        def self.puppetserver_confdir(puppet_confdir)
+          File.join(File.dirname(puppet_confdir), 'puppetserver')
         end
 
-        def self.default_ssldir
-          File.join(self.puppet_confdir, 'ssl')
+        def self.default_ssldir(confdir = nil)
+          confdir ||= puppet_confdir
```

Comment:
why not just make the default in the arg list be puppet_confdir?

Reply:
yeeeah, I thought about that.. it seemed kinda weird to me to call methods in the argument list, but I can change it.